module/app/app.go

```diff
@@ -411,6 +424,38 @@ func NewGravityApp(
 	)
 	app.slashingKeeper = &slashingKeeper
 
```

Comment:
Several keepers needed to be moved above the gravity keeper as a result of using ibcTransferKeeper and bech32IbcKeeper in gravity keeper.

Reply:
the only one that concerns me is the upgrade keeper, we don't have good enough tests around upgrading to know if this will break anything. We'll just have to be careful to bench test it.

Reply:
I just verified that none of upgradeKeeper's constructor arguments were affected, and that all of the usages of upgradeKeeper are still in their relative ordering. E.g. the first consumer of upgradeKeeper is ibcKeeper, the relative order of upgradekeeper.NewKeeper() and ibckeeper.NewKeeper() is the same as before, and the same is true for the remaining consumers.